benchmarks/src/main/kotlin/bfschannel/BfsChannelBenchmark.kt

```diff
@@ -0,0 +1,320 @@
+        Triple("USA-DISTANCE", "gr gz", "http://www.dis.uniroma1.it/challenge9/data/USA-road-d/USA-road-d.USA.gr.gz"),
+        Triple("LIVE-JOURNAL", "txt gz", "https://snap.stanford.edu/data/soc-LiveJournal1.txt.gz"))
+/**
+ * Iterations count for each graph
+ */
+private const val ITERATIONS = 5
+/**
+ * Number of coroutines that are used to execute bfs in parallel
+ */
+private val WORKERS = listOf(1, 4, 8, 16)
+
+/**
+ * This benchmark tests channel as a working queue, as a queue under contention.
+ * The benchmark creates or downloads graphs, then executes parallel BFS using channel as a queue, executes sequential BFS,
+ * compares the results and computes execution times.
+ */
+fun main() {
+    for ((graphName, graphType, graphUrl) in GRAPH_FILES) {
+        println("=== $graphName ===")
+        val graph = downloadOrCreateAndParseGraph(graphName, graphType, graphUrl)
+
+        val nodes = graph.size
+        val startNode = graph[0]
+
+        // warmup
+        val (_, sequentialDistances) = runSequentialBfs(startNode, nodes)
+
+        // benchmark iterations
+        val sequentialExecutionTimes = (1..ITERATIONS).map { runSequentialBfs(startNode, nodes).first }.toList()
+
+        for (workers in WORKERS) {
+            // warmup
+            runParallelBfs(startNode, nodes, workers)
+
+            // benchmark iterations
+            val parallelExecutionTimes = (1..ITERATIONS).map {
+                val (executionTime, parallelDistances) = runParallelBfs(startNode, nodes, workers)
+                check(parallelDistances == sequentialDistances)
+                executionTime
+            }.toList()
+
+            println("parallel workers count = $workers, parallel execution time = ${parallelExecutionTimes.average() / 1_000_000}ms std = ${computeStandardDeviation(parallelExecutionTimes) / 1_000_000}ms, sequential execution time = ${sequentialExecutionTimes.average() / 1_000_000}ms std = ${computeStandardDeviation(sequentialExecutionTimes) / 1_000_000}ms")
+        }
+    }
+}
+
+fun computeStandardDeviation(list : List<Long>) : Double {
+    var sum = 0.0
+    var standardDeviation = 0.0
+    for (num in list) {
+        sum += num
+    }
+    val mean = sum / list.size
+    for (num in list) {
+        standardDeviation += (num - mean).pow(2.0)
+    }
+    return sqrt(standardDeviation / list.size)
+}
+
+private fun runParallelBfs(startNode: Node, nodes: Int, workers: Int) : Pair<Long, List<Long>> {
+    val start = System.nanoTime()
+    val parallelDistances = bfsParallel(startNode, nodes, workers)
+    val end = System.nanoTime()
+
+    return Pair(end - start, parallelDistances)
+}
+
+private fun runSequentialBfs(startNode: Node, nodes: Int) : Pair<Long, List<Long>> {
+    val start = System.nanoTime()
+    val sequentialDistances = bfsSequential(startNode, nodes)
+    val end = System.nanoTime()
+
+    return Pair(end - start, sequentialDistances)
+}
+
+fun bfsSequential(startNode: Node, nodes: Int): List<Long> {
```

Comment:
Do we really need this sequential code? Why does it use `LinkedList` and not `ArrayQueue`?